### DISCUSSION TITLE
How to implement stepper to number field?

### DISCUSSION THREAD

**Discussion by skorphil** [Budibase/budibase#9659] [Help] [answered]
Hi, I'm trying to create a button, which is increasing the value in the field by 0.1

I have added a JS function in the on-click action of a button:
```
return $("Form.Fields.Number") + 0.1
```
But this led to a problem with floating point - I got numbers like 1, 1.1, 1.2, 1.3000000002, 1.40000000001, etc

Then I've tried
```
return ($("Form.Fields.Number") + 0.1).toFixed(1)
```

But it is not working... after a few clicks, I got an empty field.

Is there any way to make it work?

**Comment by melohagan** [comment, marked as answer]
Hey @skorphil 

Seems to be a fairly common gnarly JavaScript problem. I found this StackOverflow post to be helpful: https://stackoverflow.com/questions/831687/is-there-a-definitive-solution-to-javascript-floating-point-errors

In the end this binding worked out for me:
```js
return Math.round(($("New Form.Fields.MyNumber") + 0.1) * 100) / 100;
```

![inline_edit](https://user-images.githubusercontent.com/101575380/219017969-aee20981-f003-43e4-9c4d-b2dba73608c5.gif)

**Reply by skorphil** [reply]
Thank you, it works!
